elements/lisk-p2p/src/p2p_request.ts

```diff
@@ -68,10 +68,10 @@ export class P2PRequest {
 			if (!responseError && responsePacket) {
 				options.productivity.lastResponded = Date.now();
 				options.productivity.responseCounter += 1;
-				options.productivity.responseRate =
-					options.productivity.responseCounter /
-					options.productivity.requestCounter;
 			}
+			options.productivity.responseRate =
```

Comment:
Please, explain the reasoning behind.

Reply:
One of the unit test cases I came up with was failing.
The problem was that if the request failed, it would not update the responseRate; even though the responseRate should be lower. It would only update the productivity responseRate on success which is incorrect.